### PR TITLE
amend release strategy

### PR DESCRIPTION
**Pull request by cdoern**
we have been sending slack and email announcements for each z-stream we release. This should be amended to be up to the release team if an announcement is necessary. Not all releases with ~1 or 2 commits warrant multiple messages

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: docs/release-strategy.md [hunk 1]
```diff
@@ -65,7 +65,7 @@ The following are the steps for how Y-stream and Z-stream releases gets cut.
 1. Validate the release branch with an [E2E test](ci.md).
 1. Create a new release on GitHub targeting the release branch and using the previous Z-Stream tag as the previous release (e.g. `0.15.0` precedes `0.15.1`).
 1. Move the `stable` tag to the new release (note this tag is set to be deprecated on September 1st, 2024 - users should use PyPi to install the latest "stable" release)
-1. Announce release via the following:
+1. Announce the release if necessary. Not all Z-Streams necessitate an email or slack message if the number of commits is low or if the content of the commits is not consequential to most users. If warranted announce on:
```
Comment (nathan-weinberg) [inline docs/release-strategy.md:68]:
```suggestion
1. Announce the release if the Maintainer team deems it is necessary. If desired announce on:
```

**Review by nathan-weinberg — CHANGES_REQUESTED** [review]

File: docs/release-strategy.md (hunk 1 above)
Comment (cdoern) [reply docs/release-strategy.md:68]: I think a mix of both of these wordings makes sense.

Adding what is "necessary" and adding that the maintainers deem whether it is necessary or not makes sense to me

Comment (khaledsulayman) [inline docs/release-strategy.md:68]:
```suggestion
1. If changes warrant an announcement, announce via the following:
```

Comment (nathan-weinberg) [reply docs/release-strategy.md:68]: Why back ourselves into a corner unnecessarily?

Also fine with the wording @khaledsulayman suggested below

**Review by khaledsulayman — APPROVED** [review]

**Review by nathan-weinberg — APPROVED** [review]

**Pull request merged**